src/charm.py

```diff
@@ -307,6 +316,9 @@ def _create_discourse_environment_settings(self) -> typing.Dict[str, str]:
             Dictionary with all the environment settings.
         """
         database_relation_data = self._database.get_relation_data()
+
+        # The following could fail if the data is malformed.
+        # We/don't catch it because we don't want to silently fail in those cases
```

Comment:
I guess in that case the charm goes into ErrorState

Reply:
To be clear, this was an observation trying to prompt you to think whether that is as intended 🙂

Reply:
Yes, it's exactly what we would want in that situation.